onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -233,6 +233,50 @@ void MultiHeadAttentionTypeAndShapeInference(ONNX_NAMESPACE::InferenceContext& c
   }
 }
 
+void GroupQueryAttentionTypeAndShapeInference(ONNX_NAMESPACE::InferenceContext& ctx) {
+  // Output 0 has shape (batch_size, sequence_length, hidden_size)
+
+  // Q, K and V without packing:
+  //   Input 0 (query) has shape (batch_size, sequence_length, hidden_size)
+  //   Input 1 (key) has shape (batch_size, kv_sequence_length, hidden_size)
+  //   Input 2 (value) has shape (batch_size, kv_sequence_length, v_hidden_size)
+
+  // Q, K and V without packing and past (cross attention):
```

Comment:
Please update Operator spec if two formats are supported. The spec only has the first format (k and v are not transposed).